src/Factory/AppRouterFactory.php

```diff
@@ -2,34 +2,96 @@
 
 namespace App\Factory;
 
+use App\Blog\Archive\ArchiveController;
+use App\Blog\BlogController;
+use App\Blog\Post\PostController;
+use App\Blog\Tag\TagController;
 use App\Controller\AuthController;
 use App\Controller\ContactController;
 use App\Controller\SiteController;
+use App\Controller\UserController;
 use Psr\Container\ContainerInterface;
-use Yiisoft\Router\FastRoute\FastRouteFactory;
 use Yiisoft\Http\Method;
+use Yiisoft\Router\FastRoute\FastRouteFactory;
+use Yiisoft\Router\Group;
 use Yiisoft\Router\Route;
+use Yiisoft\Router\RouteCollectorInterface;
 use Yiisoft\Router\RouterFactory;
+use Yiisoft\Widget\WidgetFactory;
 use Yiisoft\Yii\Web\Middleware\ActionCaller;
 
 class AppRouterFactory
 {
     public function __invoke(ContainerInterface $container)
     {
+        WidgetFactory::initialize($container);
+
         $routes = [
             Route::get('/', new ActionCaller(SiteController::class, 'index', $container))
                 ->name('site/index'),
-            Route::methods([Method::GET, Method::POST], '/contact', new ActionCaller(ContactController::class, 'contact', $container))
-                ->name('site/contact'),
+            Route::methods(
+                [Method::GET, Method::POST],
+                '/contact',
+                new ActionCaller(ContactController::class, 'contact', $container)
+            )->name('site/contact'),
             Route::get('/test/{id:\w+}', new ActionCaller(SiteController::class, 'testParameter', $container))
                 ->name('site/test'),
-
-            Route::methods([Method::GET, Method::POST], '/login', new ActionCaller(AuthController::class, 'login', $container))
-                ->name('site/login'),
+            Route::methods(
+                [Method::GET, Method::POST],
+                '/login',
+                new ActionCaller(AuthController::class, 'login', $container)
+            )->name('site/login'),
             Route::get('/logout', new ActionCaller(AuthController::class, 'logout', $container))
                 ->name('site/logout'),
+
+            Route::get('/user[/page-{page:\d+}]', new ActionCaller(UserController::class, 'index', $container))
```

Comment:
Why are square brackets needed here?

Reply:
I probably wanted to make a list of user posts